### PR TITLE
Upgrade cloud-init to 23.2

### PR DESCRIPTION
**Pull request by rmhsawyer**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Upgrade cloud-init to 23.2

###### Change Log  

- Upgrade cloud-init to 23.2
- remove CVE patch no longer needed

###### Does this affect the toolchain?  

NO

###### Test Methodology

- Pipeline build id: [389762](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=389762&view=results)
- Local package and image build
- Azure VM deployment
- BuddyBuild: 
https://dev.azure.com/mariner-org/mariner/_build/results?buildId=390271&view=results
https://dev.azure.com/mariner-org/mariner/_build/results?buildId=390274&view=results

### CODE REVIEW

File: SPECS/cloud-init/cloud-init.spec [hunk 1]
```diff
@@ -132,21 +131,23 @@ make check %{?_smp_mflags}
...
-%{_sysconfdir}/dhcp/dhclient-exit-hooks.d/hook-dhclient
-%{_sysconfdir}/NetworkManager/dispatcher.d/hook-network-manager
 %config(noreplace) %{_sysconfdir}/cloud/templates/*
 %config(noreplace) %{_sysconfdir}/cloud/cloud.cfg
 %config(noreplace) %{_sysconfdir}/cloud/cloud.cfg.d/05_logging.cfg
 %config(noreplace) %{_sysconfdir}/systemd/system/sshd-keygen@.service.d/disable-sshd-keygen-if-cloud-init-active.conf
 %{_unitdir}/*
 %{_systemdgeneratordir}/cloud-init-generator
-/lib/udev/rules.d/66-azure-ephemeral.rules
+/usr/lib/udev/rules.d/66-azure-ephemeral.rules
```
Comment (christopherco) [inline SPECS/cloud-init/cloud-init.spec:140]: Any particular reason for this path change? Is there an upstream change linked to a move?

File: SPECS/cloud-init/cloud-init.spec [hunk 2]
```diff
@@ -132,21 +131,23 @@ make check %{?_smp_mflags}
 %dir %{_sysconfdir}/cloud/templates
 %doc %{_sysconfdir}/cloud/cloud.cfg.d/README
 %doc %{_sysconfdir}/cloud/clean.d/README
-%{_sysconfdir}/dhcp/dhclient-exit-hooks.d/hook-dhclient
-%{_sysconfdir}/NetworkManager/dispatcher.d/hook-network-manager
```
Comment (christopherco) [inline SPECS/cloud-init/cloud-init.spec:136]: Just checking, are these safe for us to remove? Is there an upstream cloud-init change associated with this removal?

**Review by christopherco — CHANGES_REQUESTED** [review]

File: SPECS/cloud-init/cloud-init.spec (hunk 2 above)
Comment (rmhsawyer) [reply SPECS/cloud-init/cloud-init.spec:136]: Those two files were removed in cloud-init 23.2 version and thus causing build errors. See related PR that removes those files: https://github.com/canonical/cloud-init/commit/2ee614aa6431f4339f48ebabdf2de4ebcc2401ab

File: SPECS/cloud-init/cloud-init.spec (hunk 1 above)
Comment (rmhsawyer) [reply SPECS/cloud-init/cloud-init.spec:140]: I caught this when I encountered a building error that states this file does not exist. I checked the upstream and saw that there was a recent change on default path of udev/rules (see PR: https://github.com/canonical/cloud-init/commit/70715125f3af118ae242770e61064c24f41e9a02). Change the path from /lib to /usr/lib will resolve this issue.

**Review by christopherco — APPROVED** [review]

**Pull request merged**